src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +159,175 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
     }
 }
 
+void IInferRequestInternal::SetBlobs(const std::string& name, const std::vector<Blob::Ptr>& blobs) {
+    bool all_memory = std::all_of(blobs.begin(), blobs.end(), [](const Blob::Ptr& item) {
+        return item && item->is<MemoryBlob>() && !item->is<RemoteBlob>();
+    });
+    OPENVINO_ASSERT(all_memory,
+                    "set_input_tensors/set_tensors error. Default implementation support only local memory tensors");
+
+    if (blobs.size() == 1) {
+        SetBlob(name, blobs[0]);
+        return;
+    }
+
+    checkBatchedBlobs(name, blobs);
+
+    SetBlobsImpl(name, std::make_shared<BatchedBlob>(blobs));
+}
+
+void IInferRequestInternal::SetBlobsImpl(const std::string& name, const BatchedBlob::Ptr& batched_blob) {
+    IE_THROW(NotImplemented) << "set_input_tensors/set_tensors are not supported by this plugin";
+}
+
+void IInferRequestInternal::checkBatchedBlobs(const std::string& name, const std::vector<Blob::Ptr>& blobs) {
+    OPENVINO_ASSERT(!blobs.empty(),
+                    "set_input_tensors/set_tensors can't be called with empty blobs for input '",
+                    name,
+                    "'");
```

Comment:
should not we also check that `blobs.size() != 1` because this case should be handled via SetBlob

Reply:
smth like blobs.size() == _inputs[name].getDims[BATCH_DIM] ?